src/opentime/rationalTime.cpp

```diff
@@ -94,6 +94,69 @@ is_dropframe_rate(double rate)
     return std::find(b, e, rate) != e;
 }
 
+static bool
+parseFloat(char const* pCurr, char const* pEnd, bool allow_negative, double* result)
+{
+    if (pCurr >= pEnd || !pCurr) {
+        *result = 0.0;
+        return false;
+    }
+
+    double ret = 0.0;
+    double sign = 1.0;
+
+    if (*pCurr == '+')
+        ++pCurr;
+    else if (*pCurr == '-') {
+        if (!allow_negative) {
+            *result = 0.f;
+            return false;
+        }
+        sign = -1.0;
+        ++pCurr;
+    }
+
+    // get integer part
+    uint32_t uintPart = 0;
+    while (pCurr < pEnd) {
+        char c = *pCurr;
+        if (c < '0' || c > '9')
+            break;
+        uintPart = uintPart * 10 + c - '0';
+        ++pCurr;
```

Comment:
shouldn't it also be a error here if `uintPart` this overflows?

Reply:
That had not occurred to me. I added an overflow check